librocksdb_sys/crocksdb/c.cc

```diff
@@ -3733,6 +3733,11 @@ const char* crocksdb_compactionfiltercontext_start_key(
   return result.data();
 }
 
+int crocksdb_compactionfiltercontext_reason(
```

Comment:
I used it because the usage of `int` versus the usage of enum is at least 50/50, e.g. `should_filter_table_file_creation`, `get_compression_type` uses int. Using int saves us from creating (and maintaining) another enum in c.h, on the other hand using enum in c.h doesn't benefit code checking.

Reply:
It's not about convenience but about safety. A compiler is free to choose whatever type that can store all the variants of an enum, which might not be int.

Reply:
We use static cast to cast C++ enum to int, this is safe. As long as we properly define the rust enum (`repr(i32)` or `repr(u32)`), it will be safe to map the int value to rust side enum. On the other hand, when using enum in C, you also have to carefully define the rust enum (`repr(C)`), the maintainence overhead is not smaller but greater.

Reply:
> As long as we properly define the rust enum (repr(i32) or repr(u32)), it will be safe to map the int value to rust side enum. 

However, it's defined as `repr(C)` on the Rust side.

Reply:
Yes, this would need a full sweep clean up.

Reply:
Fixed in https://github.com/tikv/rust-rocksdb/pull/743